eclipse-che/che-operator
# feat: Traefik plugin

### What does this PR do?
updates deployment of `nativeUserMode` with using traefik plugin to rewrite authorization header

### Screenshot/screencast of this PR



### What issues does this PR fix or reference?

https://github.com/eclipse/che/issues/20070


### How to test this PR?

1. deploy che with devworkspaces enabled, che-operator templates from this PR (need new env value RELATED_IMAGE_single_host_gateway_native_user_mode) and CheCluster patch:
```
spec:
  auth:
    nativeUserMode: true
```
2. check that gateway has no header-rewrite-proxy container, it should have 4 containers
3. check configmap `che-gateway-config-header-rewrite-plugin` that should contain `headerRewrite.go` with plugin code
4. every route configmap should containe additional middleware configuring this plugin
5. test starting the devworkspace or whatever. Without the plugin working, you should get 403 on loading the dashboard and kubernetes proxy of che-server won't work as it relies on correct Authorization header


### PR Checklist
          
[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
